libs/SalesforceAnalytics/SalesforceAnalytics/Classes/Logger/SFSDKLogger.m

```diff
@@ -81,25 +83,90 @@ - (instancetype)initWithComponent:(NSString *)componentName {
         self.componentName = componentName;
         self.logger = [[DDLog alloc] init];
         self.fileLogger = [[SFSDKFileLogger alloc] initWithComponent:componentName];
-        DDLogLevel logLevel = DDLogLevelError;
-#ifdef DEBUG
-        logLevel = DDLogLevelDebug;
-#endif
         DDTTYLogger *consoleLogger = [DDTTYLogger sharedInstance];
         consoleLogger.colorsEnabled = YES;
-        [self.logger addLogger:consoleLogger withLevel:logLevel];
-        [self.logger addLogger:self.fileLogger withLevel:logLevel];
+        [self.logger addLogger:consoleLogger withLevel:self.logLevel];
+        if (self.fileLoggingEnabled) {
+            [self.logger addLogger:self.fileLogger withLevel:self.logLevel];
+        }
     }
     return self;
 }
 
+- (void)setFileLoggingEnabled:(BOOL)loggingEnabled {
+    BOOL curPolicy = [self readFileLoggingPolicy];
+    [self storeFileLoggingPolicy:loggingEnabled];
+    BOOL newPolicy = [self readFileLoggingPolicy];
+
+    // Adds or removes the file logger depending on the change in policy.
+    if (curPolicy != newPolicy) {
+        if (newPolicy) {
+            [self.logger addLogger:self.fileLogger withLevel:self.logLevel]; // Disabled to enabled.
+        } else {
+            [self.logger removeLogger:self.fileLogger]; // Enabled to disabled.
+        }
+    }
+}
+
+- (BOOL)isFileLoggingEnabled {
+    return [self readFileLoggingPolicy];
+}
+
+- (DDLogLevel)getLogLevel {
+    return [self readLogLevel];
+}
+
 - (void)setLogLevel:(DDLogLevel)logLevel {
+    [self storeLogLevel:logLevel];
     [self.logger removeAllLoggers];
     DDTTYLogger *consoleLogger = [DDTTYLogger sharedInstance];
     consoleLogger.colorsEnabled = YES;
     [self.logger addLogger:consoleLogger withLevel:logLevel];
     [self.logger addLogger:self.fileLogger withLevel:logLevel];
-    _logLevel = logLevel;
+}
+
+- (void)storeFileLoggingPolicy:(BOOL)enabled {
+    @synchronized (self) {
+        NSUserDefaults *defs = [NSUserDefaults standardUserDefaults];
```

Comment:
I don't think we can move that stuff to `SalesforceAnalytics` because they depend on `SFUserAccount`. The analytics library does not have any knowledge of user accounts.